mllib/src/main/scala/org/apache/spark/ml/tree/impl/RandomForest.scala

```diff
@@ -1107,9 +1108,11 @@ private[spark] object RandomForest extends Logging {
         mutableTreeToNodeToIndexInfo
           .getOrElseUpdate(treeIndex, new mutable.HashMap[Int, NodeIndexInfo]())(node.id)
           = new NodeIndexInfo(numNodesInGroup, featureSubset)
+        numNodesInGroup += 1
+        memUsage += nodeMemUsage
+      } else {
+        flag = false
```

Comment:
"flag" isn't that descriptive. What about simply clearing the stack to make the loop terminate naturally? or a break?

Actually can the loop just use `nodeStack.pop()` at the top? it isn't necessary to peek at it with top and then later decide to pop. It can be removed no matter what.